crates/speedy-transform/src/web_transform/remove_effect.rs

```diff
@@ -0,0 +1,119 @@
+use std::collections::HashSet;
+
+use swc_common::{util::take::Take, Mark};
+use swc_ecma_ast::{BlockStmt, Expr, Id, Module, ModuleDecl, ModuleExportName, ModuleItem};
+use swc_ecma_transforms::resolver;
+use swc_ecma_visit::{VisitMut, VisitMutWith};
+
+use crate::types::TransformConfig;
+
+struct RmUseEffect {
+  use_effect_mark: Option<Id>, // used for remove useEffect()
+  react_mark: Option<Id>,      // used for remove React.useEffect()
+}
+
+const USE_EFFECT_STR: &str = "useEffect";
+
+impl VisitMut for RmUseEffect {
+  fn visit_mut_block_stmt(&mut self, n: &mut BlockStmt) {
+    let mut rm_idx_set = HashSet::new();
+    for (idx, stmt) in n.stmts.iter().enumerate() {
+      if let Some(Some(call_expr)) = stmt.as_expr().map(|expr_stmt| expr_stmt.expr.as_call()) {
+        call_expr.callee.as_expr().and_then(|callee| {
+          match callee.as_ref() {
+            Expr::Member(member) => {
+              // check React.useEffect call
+              member.obj.as_ref().as_ident().and_then(|obj_ident| {
+                member.prop.as_ident().and_then(|prop_ident| {
+                  if self.react_mark.eq(&Some(obj_ident.to_id()))
+                    && prop_ident.as_ref() == USE_EFFECT_STR
+                  {
+                    rm_idx_set.insert(idx);
+                  };
+                  Some(())
+                })
+              });
+            }
+            Expr::Ident(ident) => {
+              // check useEffect call
+              if self.use_effect_mark.eq(&Some(ident.to_id())) {
+                rm_idx_set.insert(idx);
+              }
+            }
+            _ => {}
+          };
+          Some(())
+        });
+      }
+    }
+    if !rm_idx_set.is_empty() {
+      n.stmts = n
+        .stmts
+        .take()
+        .into_iter()
+        .enumerate()
+        .filter_map(|(idx, stmt)| {
+          if rm_idx_set.contains(&idx) {
+            None
+          } else {
+            Some(stmt)
+          }
+        })
+        .collect();
+    }
+    n.stmts.visit_mut_with(self);
+  }
+}
+
+pub fn remove_call(module: &mut Module, config: &TransformConfig) {
+  if config.remove_use_effect.is_none() || !config.remove_use_effect.unwrap() {
+    return;
+  }
+
+  swc_common::GLOBALS.set(&swc_common::Globals::new(), || {
```

Comment:
Globals of a compiler need to be singleton, as it stores some meta data like filename / sourcemap locations on it. You need to make sure the AST is clean, and call with
```rust
GLBOALS.set(compiler.globals(), || {

})
// or
compiler.run(|| { // this is a sugar for the previous approach

})
```

Reply:
You also need the approach to handle type identifier resolving, I think it looks good to me for this PR and solve this issue in next PR.

Reply:
fix